lib/Model/events.js

```diff
@@ -298,15 +465,20 @@ function testPatternFn(pattern, patternSegments) {
   };
 }
 
+/**
+ * @param {Array<string | number>} segments
+ */
 function stripRestWildcard(segments) {
   // ['example', '**'] -> ['example']; return true
   var lastIndex = segments.length - 1;
-  if (segments[lastIndex] === '**') {
+  var lastSegment = segments[lastIndex];
+  if (lastSegment === '**') {
     segments.pop();
     return true;
   }
   // ['example', 'subpath**'] -> ['example', 'subpath']; return true
-  var match = /^([^\*]+)\*\*$/.exec(segments[lastIndex]);
+  if (typeof lastSegment !== 'string') return false;
+  var match = /^([^\*]+)\*\*$/.exec(lastSegment);
```

Comment:
Since the code apparently treats `segment**` and `segment.**` the same today, Let's not do this in the new signature. We can require you to make '*' and '**' separate strings.